curator/utils.py

```diff
@@ -1599,18 +1599,21 @@ def restore_check(client, index_list):
     :arg client: An :class:`elasticsearch.Elasticsearch` client object
     :arg index_list: The list of indices to verify having been restored.
     """
-    try:
-        response = client.indices.recovery(index=to_csv(index_list), human=True)
-    except Exception as e:
-        raise exceptions.CuratorException(
-            'Unable to obtain recovery information for specified indices. '
-            'Error: {0}'.format(e)
-        )
-    # This should address #962, where perhaps the cluster state hasn't yet
-    # had a chance to add a _recovery state yet, so it comes back empty.
-    if response == {}:
-        logger.info('_recovery returned an empty response. Trying again.')
-        return False
+    response = {}
+    for chunk in chunk_index_list(index_list):
+        try:
+            chunk_response = client.indices.recovery(index=chunk, human=True)
+        except Exception as e:
+            raise exceptions.CuratorException(
+                'Unable to obtain recovery information for specified indices. '
+                'Error: {0}'.format(e)
+            )
+        # This should address #962, where perhaps the cluster state hasn't yet
+        # had a chance to add a _recovery state yet, so it comes back empty.
+        if chunk_response == {}:
+            logger.info('_recovery returned an empty response. Trying again.')
+            return False
```

Comment:
Though highly unlikely, I worry that `return False` could lead to a bad loop (imagine a scenario where the first pass yields a good `chunk_response`, but a second one for whatever reason yields an empty `{}` chunk response). If that is a result of a changes during/before/after the restore, it might just get stuck bouncing back to the function that calls to see if everything restored. 

I guess we'll have to cross that bridge if we ever come to it as I do not think it easy to simulate with either a unit or integration test.